src/main/webapp/3Dscript.js

```diff
@@ -0,0 +1,183 @@
+import {FlyControls} from 'https://threejs.org/examples/jsm/controls/FlyControls.js';
+import {OrbitControls} from 'https://threejs.org/examples/jsm/controls/OrbitControls.js';
+import {GUI} from 'https://threejs.org/examples/jsm/libs/dat.gui.module.js';
+
+// These are global objects.
+let scene;
+let camera;
+let renderer;
+let clock;
+let controls;
+let orientation;
+let trajectory;
+let dataLength;
```

Comment:
Just use `pose.length` where you need it, so you don't have to worry about these two variables getting out of sync.